doc/source/serve/tutorials/asynchronous-inference/content/README.md

```diff
@@ -0,0 +1,400 @@
+
+```python
+pdf_url = "https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf"
+
+response = requests.post(
+    f"{BASE_URL}/process",
+    json={
+        "pdf_url": pdf_url,
+        "max_summary_paragraphs": 2
+    }
+)
+
+task_data = response.json()
+task_id = task_data["task_id"]["id"]
+
+print(f"✓ Task submitted!")
+print(f"  Task ID: {task_id}")
+print(f"  Status: {task_data['status']}")
+```
+
+#### 2. Poll for Task Status
+
+Check the task status to see if it's complete:
+
+```python
+# Check status (may need to run this cell multiple times)
+response = requests.get(f"{BASE_URL}/status/{task_id}")
+status_data = response.json()
+
+print(f"Task status: {status_data['status']}")
+
+if status_data['status'] == 'SUCCESS':
+    result = status_data['result']
+    print(f"\n✓ Complete!")
+    print(f"  Pages: {result['page_count']}")
+    print(f"  Words: {result['word_count']}")
+    print(f"  Time: {result['processing_time_seconds']}s")
+elif status_data['status'] == 'FAILURE':
+    print(f"\n✗ Failed: {status_data.get('error')}")
+else:
+    print(f"  Still processing... (Status: {status_data['status']})")
+```
+
+#### 3. Wait for Completion
+
+Or use a polling loop to automatically wait:
+
+```python
+# Submit a new task and wait for completion
+pdf_url = "https://arxiv.org/pdf/1706.03762.pdf"
+
+response = requests.post(
+    f"{BASE_URL}/process",
+    json={"pdf_url": pdf_url, "max_summary_paragraphs": 3}
+)
+
+task_id = response.json()["task_id"]["id"]
+print(f"Task submitted: {task_id}\n")
+
+# Poll until complete
+max_attempts = 40
+for attempt in range(max_attempts):
+    response = requests.get(f"{BASE_URL}/status/{task_id}")
+    status_data = response.json()
+    
+    if status_data['status'] == 'SUCCESS':
+        result = status_data['result']
+        print(f"\n✓ Complete!")
+        print(f"  Pages: {result['page_count']}")
+        print(f"  Words: {result['word_count']}")
+        print(f"  Time: {result['processing_time_seconds']}s")
+        print(f"\n  Summary preview:")
+        print(f"  {result['summary'][:200]}...")
+        break
+    elif status_data['status'] == 'FAILURE':
+        print(f"✗ Failed: {status_data.get('error')}")
+        break
+    elif attempt % 5 == 0:
+        print(f"  Still processing... ({status_data['status']})")
+    
+    time.sleep(3)
+```
+
+## Architecture Overview
+
+```
+┌─────────────┐
+│   Client    │
+└──────┬──────┘
+       │ HTTP POST /process
+       ▼
+┌─────────────────────┐
+│   AsyncPDFAPI       │ ← Ingress Deployment
+│ (HTTP Endpoints)    │
+└──────┬──────────────┘
+       │ enqueue_task()
+       ▼
+┌─────────────────────┐
+│   Redis Queue       │ ← Message Broker
+│ (Celery Backend)    │
+└──────┬──────────────┘
+       │ consume tasks
+       ▼
+┌─────────────────────┐
+│   PDFProcessor      │ ← Task Consumer Deployment
+│ @task_consumer      │   (Scaled to N replicas)
+│ - process_pdf       │
+└─────────────────────┘
+```
+
+## Key Concepts
```

Comment:
these concepts need to be explain inline in place where they are introduced. Keep the explanation crisp and short for more details link to a section in ray io docs